### PR TITLE
DEV-256 | upgrade cookbooks to latest versions

### PR DESCRIPTION
**Pull request by phuonglm**

### CODE REVIEW

**Comment by hoatle** [comment]
mysql not working:

vagrant@vagrant-ubuntu-trusty-64:~$ mysql -u root -pteracy

Warning: Using a password on the command line interface can be insecure.

ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

pls make sure to check all supported services should work when being enabled.

**Comment by phuonglm** [comment]
Try with this
```
root@vagrant-ubuntu-trusty-64:/home/vagrant# mysql -u root -pteracy -h 127.0.0.1
Warning: Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 3
Server version: 5.6.27-0ubuntu0.14.04.1 (Ubuntu)

Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show databases
    -> ;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
+--------------------+
3 rows in set (0.02 sec)
```

**Comment by hoatle** [comment]
make sure it should follow the docs http://dev.teracy.org/docs/develop/databases_guide.html#mysql

php --version also has problem, need to check other services, too.

**Comment by hoatle** [comment]
still problems:

- could not login into phpmyadmin
- could not remote access to mysql:
ERROR 1130 (HY000): Host '10.0.2.2' is not allowed to connect to this MySQL server

File: vagrant_config.json [hunk 1]
```diff
@@ -102,7 +102,7 @@
   "vb":{ //virtualbox settings from https://www.virtualbox.org/manual/ch08.html#vboxmanage-modifyvm
     //"gui":true,
     //"name":"teracy-dev",
-    "memory":1024,
+    "memory":2024,
```
Comment (hoatle) [inline vagrant_config.json]: not sure why need to to increase memory?

Comment (phuonglm) [inline vagrant_config.json]: I got many failed when compile PHP with 1GB ram. So bump it to 2GB

**Comment by hoatle** [comment]
failed to remote access mongodb:

```
$ mongo
MongoDB shell version: 2.4.9
connecting to: test
Wed Dec 23 03:56:41.104 Socket recv() errno:104 Connection reset by peer 127.0.0.1:27017
Wed Dec 23 03:56:41.104 SocketException: remote: 127.0.0.1:27017 error: 9001 socket exception [RECV_ERROR] server [127.0.0.1:27017] 
Wed Dec 23 03:56:41.104 DBClientCursor::init call() failed
Wed Dec 23 03:56:41.105 Error: DBClientBase::findN: transport error: 127.0.0.1:27017 ns: admin.$cmd query: { whatsmyuri: 1 } at src/mongo/shell/mongo.js:145
exception: connect failed
```

**Comment by hoatle** [comment]
`note: avoid overriding vendor cookbooks attribute on dev cookbook, should use json config instead, like this: https://github.com/teracyhq/dev/blob/develop/vagrant_config.json#L122`

for example:

```
         node.override['mongodb']['version'] = node['teracy-dev']['mongodb']['version']
         node.override['mongodb']['install_method'] = 'mongodb-org'
         node.override['mongodb']['config']['bind_ip'] = '0.0.0.0'
```

and need to find and fix in all other places.

**Comment by hoatle** [comment]
There are many many places that have `node.override` and we should use json config instead. pls check all the places.

**Comment by hoatle** [comment]
don't need anymore

**Pull request closed without merging**